integrations/cog_riffusion.py

```diff
@@ -0,0 +1,181 @@
+            device=self.device,
+            local_files_only=True,
+        )
+
+    def predict(
+        self,
+        prompt_a: str = Input(description="The prompt for your audio", default="funky synth solo"),
+        denoising: float = Input(
+            description="How much to transform input spectrogram",
+            default=0.75,
+            ge=0,
+            le=1,
+        ),
+        prompt_b: str = Input(
+            description="The second prompt to interpolate with the first,"
+            "leave blank if no interpolation",
+            default=None,
+        ),
+        alpha: float = Input(
+            description="Interpolation alpha if using two prompts."
+            "A value of 0 uses prompt_a fully, a value of 1 uses prompt_b fully",
+            default=0.5,
+            ge=0,
+            le=1,
+        ),
+        num_inference_steps: int = Input(
+            description="Number of steps to run the diffusion model", default=50, ge=1
+        ),
+        seed_image_id: str = Input(
+            description="Seed spectrogram to use", default="vibes", choices=SEED_IMAGES
+        ),
+    ) -> Output:
+        """
+        Runs riffusion inference.
+        """
+        # Load the seed image by ID
+        init_image_path = Path(SEED_IMAGES_DIR, f"{seed_image_id}.png")
+        if not init_image_path.is_file():
+            return Output(error=f"Invalid seed image: {seed_image_id}")
+        init_image = PIL.Image.open(str(init_image_path)).convert("RGB")
+
+        # fake max ints
+        seed_a = np.random.randint(0, 2147483647)
+        seed_b = np.random.randint(0, 2147483647)
+
+        start = PromptInput(prompt=prompt_a, seed=seed_a, denoising=denoising)
+        if not prompt_b:  # no transition
+            prompt_b = prompt_a
+            alpha = 0
+        end = PromptInput(prompt=prompt_b, seed=seed_b, denoising=denoising)
+        riffusion_input = InferenceInput(
+            start=start,
+            end=end,
+            alpha=alpha,
+            num_inference_steps=num_inference_steps,
+            seed_image_id=seed_image_id,
+        )
+
+        # Execute the model to get the spectrogram image
+        image = self.model.riffuse(riffusion_input, init_image=init_image, mask_image=None)
+
+        # Reconstruct audio from the image
+        params = SpectrogramParams()
+        converter = SpectrogramImageConverter(params=params, device=self.device)
+        segment = converter.audio_from_spectrogram_image(image)
+
+        if not os.path.exists("out/"):
+            os.mkdir("out")
+
+        out_img_path = "out/spectrogram.jpg"
+        image.save("out/spectrogram.jpg", exif=image.getexif())
+
+        out_wav_path = "out/gen_sound.wav"
+        segment.export(out_wav_path, format="wav")
+
+        return Output(audio=Path(out_wav_path), spectrogram=Path(out_img_path))
+
+
+# TODO(hayk): Can we get rid of the below functions and incorporate into
+# RiffusionPipeline.load_checkpoint?
+
+
```

Comment:
@daanelson I'd love to get rid of these functions to download the model and roll it into the new `RiffusionPipeline.load_checkpoint` function I'm using on line 55. The reason these are different is because it's relying on this separate call to download the weights and passing local_files_only=True. I'm wondering a few things:

1. Why is that needed versus downloading the first time with local_files_only=False?

2. Do we need a separate cache for the model and unet, or can we add a single `cache_dir` argument to the load_checkpoint method and make it work?

Reply:
@hmartiro added responses below. I've implemented one possible way to remove said functions based on your comment, will push to a separate branch shortly. 

> Why is that needed versus downloading the first time with local_files_only=False?

We want to build the image into the cog container as opposed to downloading at inference time; the philosophy being that the container should start up as quickly as possible. This makes inference on Replicate.com fast as well. That's why I've added the image download step to the cog build process in the README. It's not strictly necessary, just quite a bit faster. 

> Do we need a separate cache for the model and unet, or can we add a single cache_dir argument to the load_checkpoint method and make it work?

Nope, we don't. That's my bad, didn't read into the internals of the hf_hub cache. I'll remove it.